scripts/install.sh

```diff
@@ -0,0 +1,197 @@
+#!/bin/sh
+#
+# This script installs the latest release of Tenzir.
+#
+
+# Treat unset variables as an error when substituting.
+set -u
+
+# Abort on error.
+set -e
+
+#
+# Helper utilities
+#
+
+# Check whether a command exists.
+check() {
+  command -v "$1" >/dev/null 2>&1
+}
+
+# Only use colors when we have a TTY.
+if [ -t 1 ]
+then
+  escape() { printf "\033[%sm" "$1"; }
+else
+  escape() { :; }
+fi
+make_bold() { escape "1;$1"; }
+green="$(make_bold 32)"
+blue="$(make_bold 34)"
+bold="$(make_bold 39)"
+normal="$(escape 0)"
+
+# Prints an "action", i.e., something the script is doing.
+# Use this to transparently inform the user about the key
+# steps this script takes.
+action() {
+  printf "${blue}==>${bold} %s${normal}\n" "$@"
+}
+
+#
+# Execution
+#
+
+# Print welcome banner.
+echo "${blue}"
+echo "                _____ _____ _   _ ________ ____  "
+echo "               |_   _| ____| \ | |__  /_ _|  _ \ "
+echo "                 | | |  _| |  \| | / / | || |_) |"
+echo "                 | | | |___| |\  |/ /_ | ||  _ < "
+echo "                 |_| |_____|_| \_/____|___|_| \\_\\"
+echo "${normal}"
+echo "                            INSTALLER"
+echo
+
+# Detect OS.
+action "Identifying platform"
+platform=
+os=$(uname -s)
+if [ "${os}" = "Linux" ]
+then
+    if [ -f "/etc/debian_version" ]
+    then
+        platform=Debian
+    elif [ -f "/etc/NIXOS" ]
+    then
+        platform=NixOS
+    else
+        platform=Linux
+    fi
+elif [ "${os}" = "Darwin" ]
+then
+  platform=macOS
+fi
+echo "Found ${platform}."
```

Comment:
If platform is empty at this point then you should exit with a helpful error message.

Reply:
`uname -s` doesn't return anything empty on anything reasonable we support.

Reply:
We're dispatching on it in the next block and bail out if there's no appropriate package, so this is handled correctly.

Reply:
To program this a bit more defensively, could you add an `else` that prints a logic error and exits or an assertion here? That will make that immediately visible to the reader of this code.

Reply:
Added that now.